src/libs/actions/Welcome/OnboardingFlowTasks.ts

```diff
@@ -0,0 +1,222 @@
+import type {ValueOf} from 'type-fest';
+import type {Video} from '@libs/actions/Report';
+import {translateLocal} from '@libs/Localize';
+import CONST from '@src/CONST'
+import type {OnboardingPurpose} from '@src/CONST';
+
+type OnboardingTask = {
+    type: string;
+    autoCompleted: boolean;
+    title:
+        | string
+        | ((
+              params: Partial<{
+                  integrationName: string;
+              }>,
+          ) => string);
+    description:
+        | string
+        | ((
+              params: Partial<{
+                  adminsRoomLink: string;
+                  workspaceCategoriesLink: string;
+                  workspaceMoreFeaturesLink: string;
+                  workspaceMembersLink: string;
+                  integrationName: string;
+                  workspaceAccountingLink: string;
+                  workspaceSettingsLink: string;
+                  navatticURL: string;
+              }>,
+          ) => string);
+};
+
+type OnboardingMessage = {
+    /** Text message that will be displayed first */
+    message: string;
+
+    /** Video object to be displayed after initial description message */
+    video?: Video;
+
+    /** List of tasks connected with the message, they will have a checkbox and a separate report for more information */
+    tasks: OnboardingTask[];
+
+    /** Type of task described in a string format */
+    type?: string;
+};
+
+const selfGuidedTourTask: OnboardingTask = {
+    type: 'viewTour',
+    autoCompleted: false,
+    title: translateLocal('onboarding.tasks.selfGuidedTourTask.title'),
+    description: ({navatticURL}) => translateLocal('onboarding.tasks.selfGuidedTourTask.description', {navatticURL}),
+};
```

Comment:
Don't define any task outside the `getOnboardingMessages` function. Those won't have the correct translation on language switch.